### PR TITLE
[Artur Kostiuk]Tests for product description

### PR DESCRIPTION
**Pull request by art1kos**

### CODE REVIEW

File: src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts [hunk 1]
```diff
@@ -9,12 +11,16 @@ describe('ProductImagePreviewComponent', () => {
...
     })
-    .compileComponents();
+    .compileComponents().then(() => {
+      fixture = TestBed.createComponent(ProductImagePreviewComponent);
+      component = fixture.componentInstance;
+    });
   }));
 
+
   beforeEach(() => {
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts]: 2 beforeEach in one describe

File: src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts [hunk 2]
```diff
@@ -24,4 +30,17 @@ describe('ProductImagePreviewComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render img if smallImagesArr has length', (() => {
+    const smallImagesArr = ['http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+                            'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+                            'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+                            'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png'];
+    component.smallImagesArr = smallImagesArr;
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts]: the input property is swatches not smallImagesArr

**Review by dverbovyi — CHANGES_REQUESTED** [review]

File: src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts [hunk 3]
```diff
@@ -9,19 +11,30 @@ describe('ProductImagePreviewComponent', () => {
...
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render img if smallImagesArr has length', (() => {
+    const swatches = ['http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png'];
+    component.smallImagesArr = swatches;
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts]: you didn't get my previous comment. In order to test the component, you should only change the input properties

component.swatches = swatches;

**Review by dverbovyi — CHANGES_REQUESTED** [review]
@art1kos please solve the comment

File: src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts (hunk 3 above)
Comment (art1kos) [reply src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts]: done)

**Review by dverbovyi — APPROVED** [review]

**Pull request merged**